Improve install docs (#1760)
plone/documentation: docs/admin-guide/install-buildout.md

Create a Python virtual environment.

```shell

> python3 -m venv venv

Reviewer: Why? This requires more typing when running commands.

Author: you will lose data if you mix it like this.
Only happens since Python3, but i did see this several times when the venv was rebuild with `--clear`, which you have to do when the Python binary changed. I was using the mixed version for years, but in Python3 this is dangerous.

Reviewer: I can see how it makes it harder to delete and rebuild the virtualenv. I agree with recommending a separate venv folder.

    Plone
```

Install Buildout script in bin directory.

```shell

> ./venv/bin/buildout bootstrap

Reviewer: This isn't needed. Buildout was already installed as part of the `pip install` command on line 56.

Author: it does not install buildout, but it creates the bin/buildout script, so that you don't need to prefix the buildout run with the venv folder.all the time.

Reviewer: Right, so it's necessary (or at least convenient) if the virtualenv is in a different location.

The other option would be activating the virtualenv, but that's always a can of worms to explain to people, so let's go with your way.

```

Run Buildout.

```shell

> ./bin/buildout

Reviewer: Why? It's more typing and unnecessary.

Author: in my experience it works better than without. Sometimes on Linux calling a script without `./` will not find it.
Thats why i always use it, but I'm ok if we leave this out.

Reviewer: I don't feel strongly about it, but I don't think I ever had problems with this.

Reviewer 2: I think it is only required when trying to run a script in the current working directory, but not a binary. In other words, the following example fails:

```
> script
```

Whereas the following will succeed.

```
> ./script
```

However, as long as a path to the script is specified—whether that is `./script`, `dir1/script`, or `dir1/dir2/script`—then it will be invoked.

More [discussion from SO](https://stackoverflow.com/a/6331085/2214933) explains how `/` and `$PATH` work together.

As far as in this example, I would remove `./` because there is a `/` in `bin/buildout`, making `./` optional.

    Plone
```

> Install Buildout script in bin directory.

Reviewer 2: Because "Buildout" is the package, whereas `buildout` is the script, which tripped up both me and @davisagli.

```suggestion
Use Buildout's [`bootstrap` command](https://www.buildout.org/en/latest/topics/bootstrapping.html) to install a local `buildout` script in the {file}`bin` directory.
```

This may take a few minutes.

> When ever the buildout configuration has changed, re-run ./bin/buildout

Reviewer 2: Change to active voice, because the configuration doesn't change on its own.

```suggestion
Whenever you change the Buildout configuration, run `./bin/buildout` again.
```
